### PR TITLE
Blogpost: Writing a Kubernetes Operator/Controller

### PR DESCRIPTION
**Pull request by DmitryDodzin**

### CODE REVIEW

File: content/en/blog/kubenetes-operator/index.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+## This is where the tutorial starts
+
+In the sections that follow, we’ll be creating an APIService. We’ll use Rust, but implementations in other languages can be extrapolated from it fairly easily. 
+
+```bash
+git clone https://github.com/metalbear-co/farm-operator.git
+cd farm-operator
+```
+
+*A small note on the example structure it is broken into 3 steps each in its respective dir, each step is built upon the previous step with prebuilt images[\*](https://github.com/metalbear-co/farm-operator/pkgs/container/farm-operator).*
```
Comment (eyalb181) [inline content/en/blog/kubenetes-operator/index.md:67]:
```suggestion
*Note that the example repository is divided into three steps, each in its respective directory, and each with its prebuilt image.[\*](https://github.com/metalbear-co/farm-operator/pkgs/container/farm-operator).*
```

File: content/en/blog/kubenetes-operator/index.md [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+  group: farm.example.com
+  …
+  version: v1alpha
+```
+
+This means that when we create the APIService, Kubernetes will perform a lookup request to our operator at `/apis/farm.example.com/v1alpha` and expect it to return an APIResourceList. 
+
+This way it knows which resource requests to route to the operator. The response from the farm-operator will look like this. 
+
+{{<figure src="apiservicelist-return-value.png" alt="Json result from /apis/farm.example.com/v1alpha" height="100%" width="100%">}}
```
Comment (eyalb181) [inline content/en/blog/kubenetes-operator/index.md:156]: Can't this be a code snippet?

File: content/en/blog/kubenetes-operator/index.md [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+
+**NOTE**: *groupVersion is very important because if misconfigured, it can make Kubernetes have unexpected behavior with its built-in resources and potentially cause crashes for the entire cluster.*
+
+1. First, let’s talk about adding a new resource to be handled by the operator.
```
Comment (eyalb181) [inline content/en/blog/kubenetes-operator/index.md:160]: Add a title before this section, something like Coding our Operator

File: content/en/blog/kubenetes-operator/index.md [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+
+mirrord comes as a VS Code or IntelliJ extension, or as a CLI tool. We’ll use the CLI tool in this example.
+
+To run our operator using mirrord, we can use this command:
+
+```bash
+cargo build && mirrord exec -t deploy/farm-operator –steal ./target/debug/farm-operator
+```
+
+Our operator is now running locally, but stealing requests that are being sent to the operator deployment in the cluster!
```
Comment (eyalb181) [inline content/en/blog/kubenetes-operator/index.md:260]: Maybe add another screenshot of the local operator logging the request

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: content/en/blog/kubenetes-operator/index.md (hunk 2 above)
Comment (DmitryDodzin) [reply content/en/blog/kubenetes-operator/index.md:156]: It certainly can now that I think about it

File: content/en/blog/kubenetes-operator/index.md (hunk 4 above)
Comment (DmitryDodzin) [reply content/en/blog/kubenetes-operator/index.md:267]: Currently there are no logs in the farm-operator, probably need to add a println or something

File: content/en/blog/kubenetes-operator/index.md [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+```bash
+cargo build && mirrord exec -t deploy/farm-operator –steal ./target/debug/farm-operator
+```
+
+Our operator is now running locally, but stealing requests that are being sent to the operator deployment in the cluster! 
+
+{{<figure src="kubectl-get-llamas.png" alt="Return value from kubectl get llamas." height="100%" width="100%">}}
+{{<figure src="kubectl-describe-llamas.png" alt="Return value from kubectl describe llamas." height="100%" width="100%">}}
+
+## Using benefits of Operators
```
Comment (eyalb181) [inline content/en/blog/kubenetes-operator/index.md:278]:
```suggestion
## Using Existing Resources
```

File: content/en/blog/kubenetes-operator/index.md [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+
+Implementing APIService lets us do is to provide access to existing resources but modify or enrich them before returning them to the user. All this without some complex synchronisation because you can rely on the Kubernetes as your source of truth and act accordingly.
```
Comment (eyalb181) [inline content/en/blog/kubenetes-operator/index.md:280]:
```suggestion
One thing an APIService lets us do is provide access to existing resources, but modify or enrich them before returning them to the user. In this case, Kubernetes manages persistence for these resources, and all our operator has to do is retrieve them using the Kubernetes API. 
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

**Pull request merged**